about/about-kubernetes-networking.md

```diff
@@ -0,0 +1,79 @@
+
+Kubernetes Services can also define how a service accessed from outside of the cluster, for example using
+- a node port, where the service can be accessed via a specific port on every node
+- or a load balancer, whether a network load balancer provides a virtual IP address that the service can be accessed via from outside the cluster.
+
+### Kubernetes DNS
+
+Each Kubernetes cluster has provides a DNS service. Every pod and every service is discoverable through the Kubernetes DNS service.
+
+For example:
+- Service: `my-svc.my-namespace.svc.cluster-domain.example`
+- Pod: `pod-ip-address.my-namespace.pod.cluster-domain.example`
+- Pod created by a deployment exposed as a service: `pod-ip-address.deployment-name.my-namespace.svc.cluster-domain.example`.
+
+The DNS service is implemented as Kubernetes Service that maps to one or more DNS server pods (usually CoreDNS), that are scheduled just like any other pod. Pods in the cluster are configured to use the DNS service, with a DNS search list that includes the pod's own namespace and the cluster's default domain.
+
+This means that if there is a service named `foo` in Kubernetes namespace `bar`, then pods in the same namespace can access the service as `foo`, and pods in other namespaces can access the service as `foo.bar`
+
+Kubernetes supports a rich set of options for controlling DNS in different scenarios. You can read more about these in the Kubernetes guide {% include open-new-window.html text='DNS for Services and Pods' url='https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/' %}.
+
+### NAT outgoing
+
+The Kubernetes network model specifies that pods must be able to communicate with each other directly using pod IP addresses. But it does not mandate that pod IP addresses are routable beyond the boundaries of the cluster. Many Kubernetes network implementations use [overlay networks]({{site.baseurl}}/about/about-networking#overlay-networks). Typically for these deployments, when a pod initiates a connection to an IP address outside of the cluster, the node hosting the pod will SNAT (Source Network Address Translation) map the source address of the packet from the pod IP to the node IP. This enables the connection to be routed across the rest of the network to the destination (because the node IP is routable). Return packets on the connection are automatically mapped back by the node replacing the node IP with the pod IP before forwarding the packet to the pod.
```

Comment:
Not required for approval, but I prefer keeping lines to at most a couple of hundred characters - it makes it easier to attach comments to the specific subject.

Reply:
Done